chef/cookbooks/network/recipes/default.rb

```diff
@@ -398,33 +398,47 @@ def get_datapath_id_for_ovsbridge(bridge)
 
   # Do some MTU checks/configuration for the parent of the vlan nic
   if nic.is_a?(Nic::Vlan)
-    # 1) validate that the parent of vlan nic is not used by a network directly
-    # and the requested mtu for the parent is lower than the vlan nic mtu.
-    # That would not work and setting the mtu on the vlan would fail.
-    networks_using_parent = if_mapping.select { |net, ifaces| ifaces.include? nic.parent }.keys
-    networks_using_parent.each do |net_name|
-      net = Barclamp::Inventory.get_network_by_type(node, net_name)
-      unless net.use_vlan
-        nic_parent = Nic.new nic.parent
-        if nic_parent.mtu.to_i < ifs[nic.name]["mtu"].to_i
-          msg = "#{nic.name} wants mtu #{ifs[nic.name]["mtu"]} but network #{net_name} " \
-                "using the parent nic #{nic.parent} wants a lower mtu #{net.mtu}. " \
-                "This network mtu configuration is invalid."
+    if nic.parent
+      parent_nic = Nic.new(nic.parent)
+      Chef::Log.info("Checking MTU for nic #{nic.name} with parent #{parent_nic.name}")
+      if parent_nic.is_a?(Nic::Bond)
+        # parent is a bond
+        # check bond slaves interfaces for mtu values
+        parent_nic.slaves.each do |bond_iface|
+          next unless ifs[nic.name].key?("mtu") &&
+              ifs[bond_iface.name] &&
+              ifs[bond_iface.name].key?("mtu") &&
+              ifs[bond_iface.name]["mtu"] < ifs[nic.name]["mtu"]
+          msg = "Interface #{bond_iface.name}, part of bond #{parent_nic.name} has a "\
+               "custom mtu of #{ifs[bond_iface.name]["mtu"]} but children vlan #{nic.name} "\
+               "wants a mtu of #{ifs[nic.name]["mtu"]}. This configuration is invalid."
```

Comment:
(Basically just repeating my earlier comment, as it got hidden by the latest changes)
@nvin-krja Can you explain when exactly this can happen? I'd be a little surprised if crowbar sets up a bond on interfaces that has a different mtu than the bond itself. So I'd like to find out if we really need a special case.